packages/react-native-codegen/src/parsers/parsers-primitives.js

```diff
@@ -521,36 +521,63 @@ function emitCommonTypes(
   nullable: boolean,
   parser: Parser,
 ): $FlowFixMe {
-  const genericTypeAnnotationName =
-    parser.nameForGenericTypeAnnotation(typeAnnotation);
-
-  const typeMap = {
-    Stringish: emitStringish,
-    Int32: emitInt32,
-    Double: emitDouble,
-    Float: emitFloat,
-    UnsafeObject: emitGenericObject,
-    Object: emitGenericObject,
-    $Partial: emitPartial,
-    Partial: emitPartial,
-  };
+  const typeAnnotationName = parser.convertKeywordToTypeAnnotation(
+    typeAnnotation.type,
+  );
 
-  const emitter = typeMap[genericTypeAnnotationName];
-  if (!emitter) {
-    return null;
-  }
+  switch (typeAnnotationName) {
+    case 'BooleanTypeAnnotation': {
+      return emitBoolean(nullable);
+    }
+    case 'NumberTypeAnnotation': {
+      return emitNumber(nullable);
+    }
+    case 'VoidTypeAnnotation': {
+      return emitVoid(nullable);
+    }
+    case 'StringTypeAnnotation': {
+      return emitString(nullable);
+    }
```

Comment:
can't these be added to the type map? 🤔 we can actually do something like:
```ts
const typeAnnotationName =  parser.convertKeywordToTypeAnnotation(
    typeAnnotation.type,
  );
const genericTypeAnnotationName = parser.nameForGenericTypeAnnotation(typeAnnotation);

const simpleEmitter = typeMap[typeAnnotationName] 
if (!simpleEmitter) {
  return simpleEmitter(nullable)
} 

const emitter = typeMap[genericTypeAnnotationName];
      if (!emitter) {
        return null;
      }

      return emitter(
        nullable,
        hasteModuleName,
        typeAnnotation,
        types,
        aliasMap,
        enumMap,
        tryParse,
        cxxOnly,
        parser,
      );
```

I think it will simplify the code a bit.
How do you feel?

Reply:
Sure @cipolleschi, we can definitely simplify the code with that. I've just pushed a commit updating this